src/PointEval/point_values.jl

```diff
@@ -1,47 +1,43 @@
 # Optimized version of PointScalarValues which avoids i) recomputation of dNdξ and
 # ii) recomputation of dNdx. Only allows function evaluation (no gradients) which is
 # what is used in get_point_values.
-struct PointScalarValuesInternal{dim,T<:Real,refshape<:AbstractRefShape} <: CellValues{dim,T,refshape}
-    N::Vector{T}
+struct PointValuesInternal{dim,T<:Real,refshape<:AbstractRefShape,N_t} <: CellValues{dim,T,refshape}
+    N::Vector{N_t}
```

Comment:
cc @koehlerson we need to make a PR in Ferrite viz to fix this. 

Fredrik: Maybe we can add an internal deprecation of this for now?

Reply:
Also some docs regarding the invariants between $N_t$ and $T$ might be nice. I think I am not really understanding the motivation here anyway.

Reply:
This basically is https://github.com/Ferrite-FEM/Ferrite.jl/issues/682 for this implementation. Previously the vectorization was done on the outside, with awkward reshapes for vector valued functions.

Reply:
Why not constraint T differently then? T is now unused, right?

Reply:
It is a type param of the supertype (for now), unfortunately.

Reply:
Will address in https://github.com/Ferrite-FEM/Ferrite.jl/pull/651 then after this is merged. This with PR we should also have everything inplace for the merge (after I merge new master).